backends/tfhe-cuda-backend/cuda/src/integer/integer.cuh

```diff
@@ -134,21 +138,38 @@ __host__ void integer_radix_apply_bivariate_lookup_table_kb(
     int_radix_lut<Torus> *lut) {
   cudaSetDevice(stream->gpu_index);
   // apply_lookup_table_bivariate
-
   auto params = lut->params;
+  auto pbs_type = params.pbs_type;
   auto big_lwe_dimension = params.big_lwe_dimension;
+  auto small_lwe_dimension = params.small_lwe_dimension;
+  auto ks_level = params.ks_level;
+  auto ks_base_log = params.ks_base_log;
+  auto pbs_level = params.pbs_level;
+  auto pbs_base_log = params.pbs_base_log;
+  auto glwe_dimension = params.glwe_dimension;
+  auto polynomial_size = params.polynomial_size;
+  auto grouping_factor = params.grouping_factor;
   auto message_modulus = params.message_modulus;
 
   // Left message is shifted
-  pack_bivariate_blocks(stream, lut->tmp_lwe_before_ks, lwe_array_1,
-                        lwe_array_2, lut->lwe_indexes_in, big_lwe_dimension,
-                        message_modulus, num_radix_blocks);
+  auto lwe_array_pbs_in = lut->tmp_lwe_before_ks;
+  pack_bivariate_blocks(stream, lwe_array_pbs_in, lut->lwe_trivial_indexes,
+                        lwe_array_1, lwe_array_2, lut->lwe_indexes_in,
+                        big_lwe_dimension, message_modulus, num_radix_blocks);
   check_cuda_error(cudaGetLastError());
 
   // Apply LUT
-  integer_radix_apply_univariate_lookup_table_kb(stream, lwe_array_out,
-                                                 lut->tmp_lwe_before_ks, bsk,
-                                                 ksk, num_radix_blocks, lut);
```

Comment:
Why do you replace this call here? Seems like the old code would work no?

Reply:
This is related to the bug on lwe indexes I mentioned. 

For the univariate version, we have to use `lut->lwe_indexes_in` as input indexes for the keyswitch and a trivial indexing as output ( `range(0, num_radix_block)` like), and then this trivial indexing as input of the PBS and then `lut->lwe_indexes_out` as the output.

We have to do something similar for the bivariate version but now we also have to handle the indexing for packing. `integer_radix_apply_univariate_lookup_table_kb` doesn't take as input the input/output indexing but assumes the one in `lut` is the one to be used. I thought it would be easier to just break dependency between uni/bivariate versions instead of changing that.